### PR TITLE
Fixed paths in layout.html

### LINKED ISSUE #662: 404 pages in docs
For example, while being at https://pytorch.org/ignite/master/contrib/handlers.html
and clicking on "Examples" in top header it redirects to https://pytorch.org/ignite/master/contrib/examples.html
which does not exist :(

Same happens with other header links: Concepts, Quick Start etc

cc @bosr

### PR DESCRIPTION
**Pull request by anmolsjoshi**
Fixes #662 

Description:
Corrected paths in layout.html such that header links work

Check list:
* [ ] New tests are added (if a new feature is added)
* [ ] New doc strings: description and/or example code are in RST format
* [ ] Documentation is updated (if required)

### CODE REVIEW

**Comment by vfdev-5** [comment]
@anmolsjoshi how did you tested that. Does not work locally for me. 
IMO, adding `/` to relative url will send to `https://pytorch.org/` instead of `https://pytorch.org/ignite`...

**Comment by anmolsjoshi** [comment]
@vfdev-5 closing this PR. Will have it working on my personal github pages before creating a PR

**Comment by anmolsjoshi** [comment]
@vfdev5 please have a look at this link - https://anmolsjoshi.github.io/ignite/

**Comment by vfdev-5** [comment]
@anmolsjoshi seems the bug is there. I do
-> https://anmolsjoshi.github.io/ignite/
click on contrib.mertrics link
-> https://anmolsjoshi.github.io/ignite/contrib/metrics.html
and then click on quickstart
-> https://anmolsjoshi.github.io/ignite/contrib/quickstart.html

**Comment by anmolsjoshi** [comment]
@vfdev-5 thanks, let me review

**Comment by anmolsjoshi** [comment]
@vfdev-5 it seems to work correctly for the master branch. 

Looking at this travis.yml - https://github.com/pytorch/ignite/blob/master/.travis.yml#L105

Shouldn't use-master-templates apply layout.html to every tag/branch?

**Comment by vfdev-5** [comment]
@anmolsjoshi yes I confirm that on master it does not show 404 anymore. 

> Shouldn't use-master-templates apply layout.html to every tag/branch?

yes, that should be like that. I need to check this again from my side

**Comment by anmolsjoshi** [comment]
@vfdev-5 I have found the error - its [here](https://github.com/vfdev-5/sphinxcontrib-versioning/blob/master/sphinxcontrib/versioning/routines.py#L146-L147)

except for 0.3.0, master and every other version works fine on my link - anmolsjoshi.github.io/ignite - if you repeat the steps as you described above

The reason it doesn't fail in other versions is because _templates folder did not exist in other versions of ignite. 

0.3.0 had a _templates folder. The sphinx-versioning code only copies _templates if the folder does not exist, if we make sure it overwrites existing folders - this PR will work fine

**Comment by anmolsjoshi** [comment]
@vfdev-5 have a look here - https://github.com/vfdev-5/sphinxcontrib-versioning/pull/2

**Comment by anmolsjoshi** [comment]
https://anmolsjoshi.github.io/ignite/

File: .travis.yml [hunk 1]
```diff
@@ -99,7 +99,7 @@ jobs:
 
       install:
         - pip install -r docs/requirements.txt
-        - pip install git+https://github.com/vfdev-5/sphinxcontrib-versioning.git
+        - pip install git+https://github.com/anmolsjoshi/sphinxcontrib-versioning.git
```
Comment (vfdev-5) [inline .travis.yml]:
```suggestion
        - pip install git+https://github.com/vfdev-5/sphinxcontrib-versioning.git
```

File: docs/source/_templates/layout.html [hunk 2]
```diff
@@ -90,19 +90,20 @@
       <div class="main-menu">
         <ul>
           <li>
-            <a href="{{ theme_variables.external_urls['concepts'] }}">Concepts</a>
+<!--            <link rel="index" title="{{ _('Index') }}" href="{{ pathto('genindex') }}" />-->
```
Comment (vfdev-5) [inline docs/source/_templates/layout.html]:
```suggestion
```

File: .travis.yml (hunk 1 above)
Comment (anmolsjoshi) [reply .travis.yml]: Done

**Comment by vfdev-5** [comment]
Thanks @anmolsjoshi

**Pull request merged**